contracts/staking/PoSValidatorManager.sol

```diff
@@ -350,13 +375,22 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         // the complete step, even if the delivered nonce is greater than the nonce used to
         // initialize the removal.
         delegator.status = DelegatorStatus.PendingRemoved;
-        delegator.endedAt = uint64(block.timestamp);
+        delegator.endedAt = currentTime;
```

Comment:
Do we handle the case when the validator has already previously been ended correctly here?

In that case, we shouldn't send any message to the P-chain, and the delegator should only be reward for the time period up until the validator end time.

Reply:
Accounted for in this PR https://github.com/ava-labs/teleporter/pull/554